features/sync/current_branch/feature_branch/merge_sync_strategy/no_push/enabled.feature

```diff
@@ -0,0 +1,45 @@
+Feature: no-push setting set to "true"
+
+  Background:
+    Given the current branch is a feature branch "feature"
+    And the commits
+      | BRANCH  | LOCATION | MESSAGE               |
+      | main    | local    | local main commit     |
+      |         | origin   | origin main commit    |
+      | feature | local    | local feature commit  |
+      |         | origin   | origin feature commit |
+    And Git Town setting "no-push" is "true"
+
+  Scenario: result
+    When I run "git-town sync"
+    Then it runs the commands
+      | BRANCH  | COMMAND                                 |
+      | feature | git fetch --prune --tags                |
+      |         | git checkout main                       |
+      | main    | git rebase origin/main                  |
+      |         | git push                                |
+      |         | git checkout feature                    |
+      | feature | git merge --no-edit --ff origin/feature |
+      |         | git merge --no-edit --ff main           |
+    And all branches are now synchronized
+    And feature branch is not pushed
+    And all parents of feature are pushed
+    And the current branch is still "feature"
+    And these commits exist now
+      | BRANCH  | LOCATION      | MESSAGE                                                    |
+      | main    | local, origin | origin main commit                                         |
+      |         |               | local main commit                                          |
+      | feature | local, origin | local feature commit                                       |
+      |         |               | origin feature commit                                      |
+      |         |               | Merge remote-tracking branch 'origin/feature' into feature |
+      |         |               | origin main commit                                         |
+      |         |               | local main commit                                          |
+      |         |               | Merge branch 'main' into feature                           |
+
+  Scenario: undo
+    When I run "git-town undo"
+    Then it prints:
+      """
+      nothing to undo
+      """
+    And it runs no commands
```

Comment:
Starting point for the end-to-end test for undoing this sync. As usual, it's easiest to run it, copy-and-paste the missing table content, and verify that it is indeed what you want here.

```suggestion
    Then it runs the commands
      | BRANCH   | COMMAND               |
    And the current branch is still "feature"
    And the initial commits exist
    And the initial lineage exists
```